### PR TITLE
Services My Service Navmazing

### PR DESCRIPTION
**Pull request by mshriver**
Purpose or Intent
=================

Moving myservice.py out of catalog package because My Services isn't part of catalogs.

* Adding navmazing destinations to MyService class, refactoring class methods for new navigation.
* Moved VM shutdown wait to a method to reduce duplication.

Not addressing services.test_provision_stack failures in this PR, see #3805 

{{ pytest: cfme/tests/services/test_myservice.py --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 7c74be3dd7f344ce7a9e4d587744aa9561da6d08:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5cf81b779fc3dd8481becbf99f0c45df1413c590:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b1536c0435bb3e2773453011c455fb66a19cf854:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eb881f0f911abbdcb179d5920707a3ef65f1d51a:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d862b8da8fb42e83672ccce8b599ae552c3dfa33:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3b8af91f42b60976c39730cf317b09d400bbe329:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b81c83933eb02f9d525041ae323bbed4f6ab4096:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8c779dd6a2a8aa8a59fa378788341bba3de81f62:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fab55b4c2af650d32281c7617682ca7420ca89ac:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/tests/test_ui_problems.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 TOLERANCE = 20
 # TODO: Replace nav branch strings with navmazing object destinations when supported
 LOCATIONS = [
-    "control_explorer", "automate_explorer", "automate_customization", "my_services",
+    "control_explorer", "automate_explorer", "automate_customization",
```
Comment (psav) [inline cfme/tests/test_ui_problems.py]: So, I think that a) I like this
b) did you want to start getting the code in place to do the navigation depending on the type of the destination? My idea is that you can do something like
```python
if isinstance(destination, basestring):
    force_navigate(destination):
else:
    navigate_to(destination[0], destination[1])
```
or something like that.....

Comment (psav) [inline cfme/tests/test_ui_problems.py]: I already do this is one place already so maybe we should make a consolidated function in a common area that takes either a string or a tuple and blah blah

**Comment by mshriver** [comment]
I'm expecting to see some PRT failures, will comment on the latest run when its done.

**Comment by dajoRH** [comment]
Lint report for commit a8650bcd3248cb4379d8c6418b2be270ce00c5aa:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/tests/test_ui_problems.py (hunk 1 above)
Comment (mshriver) [inline cfme/tests/test_ui_problems.py]: I like the style, and have seen/used it in some other PRs, but would like to leave it out of this PR to control scope.

**Comment by dajoRH** [comment]
Lint report for commit c01902900ea8e6926f96a8d9d1b230f0b3157554:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/services/catalogs/service_catalogs.py [hunk 2]
```diff
@@ -73,4 +74,5 @@ def order_stack_item(self, catalog, catalog_item):
                                     'catalog_item': catalog_item})
         stack_form.fill(self.stack_data)
         sel.click(form_buttons.submit)
+        wait_for(flash.get_messages, num_sec=10, delay=2)
```
Comment (mshriver) [inline cfme/services/catalogs/service_catalogs.py]: I've been hitting a failure intermittently on PRT and locally, where the assert_success_message() call logs that the messages list is empty - while the screenshot displays the message. Waiting for messages() to be populated first mitigates it, but may not be the best 'fix' here.

Could add an optional 'wait' boolean, defaulting to False, to flash.assert_success_message()

**Comment by dajoRH** [comment]
Lint report for commit 7bacaeb1972d6068a20991480f99aa647559d5ca:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/services/test_provision_stack.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from cfme.services.catalogs.catalog import Catalog
 from cfme.services.catalogs.orchestration_template import OrchestrationTemplate
 from cfme.services.catalogs.service_catalogs import ServiceCatalogs
-from cfme.services.catalogs.myservice import MyService
+from cfme.services.myservice import MyService
```
Comment (mshriver) [inline cfme/tests/services/test_provision_stack.py]: NOTE: I am not addressing test_provision_stack failures in this PR, see #3805 for changes to this test.

**Pull request merged**